Directory change

vhe182 opened the pull request:
Could you please verify and merge?  Thank you in advance for your hard work.

vhe182 committed:
Change the default name of workloads

The default name was changed to reflect the type of workload the
config file represents.

vhe182 committed:
Change filename parsing for workload clarity

String parsing was changed so that the final subdirectory name is
indicative of the type of workload that is run.  Hopefully, making
navigation of the resulting data easier.

vhe182 committed:
Add output subdirectory to clarify workload type

The subdirectory is now similar (in order) to the fio.job file
name of the workload.  It now appears thusly:

[small|large]_workload_[4,128,...]k_blksz_[1|32]_iodepth
_[0 ..100]_rwmix.fio

sample: large_workload_128k_blksz_1_iodepth_100_rwmix

besanradwan commented on utils/pkb_automated_runner.sh:
```diff
@@ -18,32 +18,36 @@ function run_pkb_provision() {
 python fio_job_file_generator.py "$CONFIG_FILE"
 for file in generated_fio_files/*; do
   run_ids=
+  subdir="$(basename $file .fio)"
```
Do you want to have "subdir=" instead of "subdir="$(basename $file .fio)" and then under line 49, define subdir? I say this because while it works, its not as consistent with the way we defined run_ids.

besanradwan replied:
Also would you be open to calling it sub_directory rather than subdir?

vhe182 replied:
I agree with your 2nd comment and have changed the variable name to be match the variable style.

comment 1 though, I disagree with.  If I move it below line 49 (within the for-loop) then we take a performance hit.  The file name doesn't change within the loop at 49 so the sub_directory variable would be updated to the same thing every loop.  I say better to avoid it and assign the value to sub_directory right below the for-loop at line 42.  Of course, maybe I am misunderstanding something about the code in this section (hope not, gulp).  If so, please let me know.

besanradwan replied:
Sounds good

besanradwan commented on utils/fio_job_file_generator.py:
```diff
@@ -63,7 +63,7 @@ def create_fio_job_files(config_file=None):
         for iodepth in config_file_params['iodepths']:
             for rwmix in config_file_params['rwmixs']:
-                filename = '%s_%s_%s_%s.fio' % (config_file_params['filename'],
+                filename = '%s_%s_blksz_%s_iodepth_%s_rwmix.fio' % (config_file_params['filename'],
```
Perhaps change blksz to blocksize?

vhe182 replied:
I agree with you.  Change made :)

besanradwan replied:
Thanks!

besanradwan commented on utils/pkb_automated_runner.sh:
```diff
@@ -18,32 +18,36 @@ function run_pkb_provision() {
   done
   for vm in $(seq 1 "$VMS"); do
-      run_pkb_benchmark "$file" "${run_ids[$vm]}" &
+      run_pkb_benchmark "$file" "${run_ids[$vm]}" "$subdir" &
```
subdir would need to be changed to sub_directory here too

vhe182 replied:
changed as well :)

besanradwan replied:
Great!

besanradwan commented on utils/pkb_automated_runner.sh:
```diff
@@ -18,32 +18,36 @@ function run_pkb_provision() {
 
 if [[ ! -d "$OUT_DIRECTORY" ]]; then
-    mkdir -p "$OUT_DIRECTORY"
+  mkdir -p "$OUT_DIRECTORY"
```
Since you added the check for the output directory on line 25, do we still need this check for the out directory?

vhe182 replied:
Good point.  With the "-p" flag, we only need the one in the run_pkb_benchmark() function.  Also, I confirmed that $OUT_DIRECTORY isn't needed before the directory check at line 25.  

Result: I will remove the directory check and make at lines 36-38

besanradwan replied:
Thanks !

vhe182 committed:
Change blksz to blocksize for subdirectory name

vhe182 committed:
Change variable name to match nameing style

"subdir" becomes "sub_directory"

vhe182 committed:
Remove unnecessary directory check

A previous commit added a directory check at another location.
Although this adds extra executions of checks due to the new
location being within another layer of for-loop.  It was confirmed
that this new location makes the original check unnecessary so it
has been removed in this commit.

besanradwan commented:
LGTM!
